### PR TITLE
Move cocina processing to a background job

### LINKED ISSUE #530: Too many queued requests causes errors in purl-fetcher client
## Error
reported from contentsearch-prod when attempting to fetch deletes:

`JSON::ParserError: 809: unexpected token at '<h2>This website is under heavy load (queue full)</h2><p>We're sorry, too many people are accessing this website at the same time. We're working on this problem. Please try again later.</p>'`

this seems to happen with some regularity, such that we get at least one email from honeybadger per day. we could make purl-fetcher_client smart enough not to try to parse the error as JSON, but we probably have some diligence due to investigate why purl-fetcher is "under heavy load".

this message apparently originates from passenger and appears when too many HTTP requests are queued by the web server. we might be able to mitigate it by turning up [`PassengerMaxRequestQueueSize`](https://www.phusionpassenger.com/library/config/apache/reference/#passengermaxrequestqueuesize), but it'd be nice to know why it's happening in the first place.

## Backtrace
    line 27 of [PROJECT_ROOT]/app/jobs/harvest_purl_fetcher_job.rb: perform_deletes
    line 10 of [PROJECT_ROOT]/app/jobs/harvest_purl_fetcher_job.rb: block in perform
    line 54 of [PROJECT_ROOT]/app/jobs/harvest_purl_fetcher_job.rb: block in with_state_file

[View full backtrace and more info at honeybadger.io](https://app.honeybadger.io/projects/54556/faults/74529416)

### PR DESCRIPTION
**Pull request by jcoyne**
Fixes #530

This has been tested on stage and handled a 17,000 item collection with no problem.

### CODE REVIEW

File: spec/consumers/purl_updates_consumer_spec.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+require 'rails_helper'
+
+RSpec.describe PurlUpdatesConsumer do
+  let(:message) { instance_double(Racecar::Message, value: message_value) }
+  # subject(:process) { described_class.nesw.process(message) }
```
Comment (corylown) [inline spec/consumers/purl_updates_consumer_spec.rb:5]: Should this commented out line be deleted?

**Review by corylown — CHANGES_REQUESTED** [review]
Thanks for working on this. One question about a commented out line.

File: spec/consumers/purl_updates_consumer_spec.rb (hunk 1 above)
Comment (jcoyne) [reply spec/consumers/purl_updates_consumer_spec.rb]: Sure, removed.

File: app/consumers/purl_updates_consumer.rb [hunk 2]
```diff
@@ -2,9 +2,10 @@ class PurlUpdatesConsumer < Racecar::Consumer
   subscribes_to "purl-update"
 
   # Currently this is a no-op until we are able to set up a systemd script to run racecar
```
Comment (thatbudakguy) [inline app/consumers/purl_updates_consumer.rb:4]: is this still true?

File: app/controllers/v1/purls_controller.rb [hunk 3]
```diff
@@ -29,15 +29,14 @@ def update
               end
 
       if params[:type]
-        # Racecar.produce_sync(value: cocina_object.to_json, key: druid_param, topic: "purl-update")
-        PurlCocinaUpdater.new(@purl, cocina_object).update
+        Racecar.produce_sync(value: cocina_object.to_json, key: druid_param, topic: "purl-update")
       else
         # This path is a fallback used until dor-services-app is providing the data we need
```
Comment (thatbudakguy) [inline app/controllers/v1/purls_controller.rb:34]: Can this still happen?

File: spec/consumers/purl_updates_consumer_spec.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+             ]
+           })
+      .to_json
+  end
+
+  before do
+    described_class.new.process(message)
+  end
+
+  it "creates a Contact for each email address in the topic" do
```
Comment (thatbudakguy) [inline spec/consumers/purl_updates_consumer_spec.rb:25]: this test title doesn't seem real

File: app/controllers/v1/purls_controller.rb (hunk 3 above)
Comment (thatbudakguy) [reply app/controllers/v1/purls_controller.rb:34]: Ah, I see #581 removes it entirely

Comment (jcoyne) [reply app/controllers/v1/purls_controller.rb:34]: Yep.

File: app/consumers/purl_updates_consumer.rb (hunk 2 above)
Comment (jcoyne) [reply app/consumers/purl_updates_consumer.rb]: No, I updated the comment.

**Review by thatbudakguy — APPROVED** [review]

**Pull request merged**